### PR TITLE
Support custom api url

### PR DESCRIPTION
**Pull request by klb3713**

### CODE REVIEW

File: src/lib/constants.ts [hunk 1]
```diff
@@ -4,4 +4,6 @@ export const APP_DESCRIPTION =
 
 export const STORAGE_KEY = 'OPENAI_API_KEY';
 
+export const STORAGE_URL = 'http://ai-api.e-tudou.com:9000/v1';
```
Comment (hkurma) [inline src/lib/constants.ts]: @klb3713 This const is used for browser storage, would like to change this key and value to 
`OPENAI_API_BASE_URL`.

**Review by hkurma — CHANGES_REQUESTED** [review]

File: src/app/page.tsx [hunk 2]
```diff
@@ -4,21 +4,29 @@ import { useRouter } from 'next/navigation';
...
+  APP_TITLE,
+  STORAGE_KEY,
+  STORAGE_URL,
+} from '@/lib/constants';
 import { ArrowUpRight } from 'lucide-react';
 import { ModeToggle } from '@/components/mode-toggle';
 
 const Index = () => {
   const [apiKey, setApiKey] = useState<string>('');
+  const [baseURL, setBaseURL] = useState<string>('');
```
Comment (hkurma) [inline src/app/page.tsx]: @klb3713 Default baseUrl to `https://api.openai.com/v1`. 

Will help users using OpenAI's default base URL.

**Review by hkurma — CHANGES_REQUESTED** [review]

File: src/app/page.tsx [hunk 3]
```diff
@@ -37,6 +45,13 @@ const Index = () => {
           value={apiKey}
           onChange={(e) => setApiKey(e.target.value)}
         />
+        <Input
```
Comment (hkurma) [inline src/app/page.tsx]: @klb3713 I would move this above API Key, since it will have a default value and user's would need to enter API Key.

**Review by hkurma — CHANGES_REQUESTED** [review]

File: src/lib/constants.ts (hunk 1 above)
Comment (klb3713) [reply src/lib/constants.ts]: yes, sorry, I forgot to modify this default, it's our internal service

**Pull request closed without merging**